### DISCUSSION TITLE
is RPC supported by the lib?

### DISCUSSION THREAD

**Discussion by fililili** [NVIDIA/stdexec#1272] [General]
What if I hope a sender run in a remove server?

**Comment by ericniebler** [comment]
There is not yet any RPC support in stdexec for any platform. I would be happy to take a pull request, and there are a lot of people who would be happy to help you get the stdexec details right, if you feel so inclined.

**Reply by fililili** [reply]
hi, @ericniebler . I have done some study. I think if we want to support RPC, we should implement a rpc_sheduler then sender can transfer to rpc_shedular easily. However, I cannot find a proper way to do that.
The main problem is, original RPC framework has to write server and client code separately. Also, original RPC framework has to register function one by one. So I cannot image how to provide a rpc_shedular. Maybe it need some compiler work.
In my understanding, CUDA is a client-server mode, too. And it's implemented by Nvidia's special compiler.
